cloudify/plugins/workflows.py

```diff
@@ -214,6 +219,15 @@ def auto_heal_reinstall_node_subgraph(
 
         ctx.refresh_node_instances()
 
+    graph = ctx.graph_mode()
+    if force_reinstall:
+        return lifecycle.reinstall_node_instances(
+            graph=graph,
+            node_instances=subgraph_node_instances,
+            related_nodes=set(ctx.node_instances) - subgraph_node_instances,
+            ignore_failure=ignore_failure,
+        )
+
```

Comment:
Because we might like to check status first and then reinstall if user insists.  I should probably even regenerate `subgraph_node_instances` before `force_reinstall`.